code/scripting/api/libs/hud.cpp

```diff
@@ -125,33 +142,57 @@ ADE_FUNC(setHUDGaugeColor, l_HUD,
 
 ADE_FUNC(getHUDGaugeColor,
 	l_HUD,
-	"number|string gaugeNameOrIndex",
-	"Color specified in the config to draw the gauge",
+	"number|string gaugeNameOrIndex, [boolean ReturnType]",
+	"Color specified in the config to draw the gauge. True to return raw rgba, false to return color object. Defaults to true.",
 	"number, number, number, number",
 	"Red, green, blue, and alpha of the gauge")
 {
 	int idx = getDefaultGaugeIndex(L);
 
+	bool rc = true;
+	ade_get_args(L, "|b", &rc);
```

Comment:
This function should be tested.  The `getDefaultGaugeIndex(L)` function already gets one argument, and the `ade_get_args` function does not pick up at the next spot in the parameter list, as one might expect.  I believe calling `ade_get_args(L, "*|b", &rc);` instead will fix it.

Reply:
Changes tested and working.